File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/Race.java
package formula.bollo.app.entity;

import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "race")
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
public class Race {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "circuit_id", referencedColumnName = "id")
    private Circuit circuit;

    @Column(name = "date_start", nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date dateStart;

    @Column(name = "finished", nullable = false, length = 1)
    private int finished;
    
    @ManyToOne
    @JoinColumn(name = "season_id", referencedColumnName = "id")
    private Season season;
}

File: Formula-Bollo/src/shared/models/race.ts
import { Circuit } from "./circuit";
import { Season } from "./season";

export class Race {
  id: number;
  circuit: Circuit;
  dateStart: Date;
  season: Season | undefined;
  finished: number;

  constructor(
    id: number,
    circuit: Circuit,
    dateStart: Date,
    finished: number,
    season?: Season,
  ) {
    this.id = id;
    this.circuit = circuit;
    this.dateStart = dateStart;
    this.finished = finished;
    this.season = season;
  }
}
